crates/configuration/src/shared/node.rs

```diff
@@ -70,6 +71,76 @@ pub struct NodeConfig {
     p2p_port: Option<Port>,
     p2p_cert_hash: Option<String>,
     db_snapshot: Option<AssetLocation>,
+    // used to skip serialization of fields with defaults to avoid duplication
+    chain_context: ChainDefaultContext,
+}
+
+impl Serialize for NodeConfig {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut state = serializer.serialize_struct("NodeConfig", 18)?;
+        state.serialize_field("name", &self.name)?;
+
+        if self.image == self.chain_context.default_image {
+            state.skip_field("image")?;
+        } else {
+            state.serialize_field("image", &self.image)?;
+        }
+
+        if self.command == self.chain_context.default_command {
+            state.skip_field("command")?;
+        } else {
+            state.serialize_field("command", &self.command)?;
+        }
+
+        if self.args.is_empty() || self.args == self.chain_context.default_args {
+            state.skip_field("args")?;
+        } else {
+            state.serialize_field("args", &self.args)?;
+        }
+
+        state.serialize_field("is_validator", &self.is_invulnerable)?;
+        state.serialize_field("is_invulnerable", &self.is_invulnerable)?;
+        state.serialize_field("is_bootnode", &self.is_bootnode)?;
+        state.serialize_field("initial_balance", &self.initial_balance)?;
+
+        if self.env.is_empty() {
+            state.skip_field("env")?;
+        } else {
+            state.serialize_field("env", &self.env)?;
+        }
+
+        if self.bootnodes_addresses.is_empty() {
+            state.skip_field("bootnodes_addresses")?;
+        } else {
+            state.serialize_field("bootnodes_addresses", &self.bootnodes_addresses)?;
+        }
+
+        state.serialize_field("resources", &self.resources)?;
+
+        if self.resources == self.chain_context.default_resources {
+            state.skip_field("resources")?;
+        } else {
+            state.serialize_field("resources", &self.resources)?;
```

Comment:
The resources here have a flat shape and in the old version are an object, I think we should change the way we serialize this field.

Reply:
I mentioned this before:
```json
{
  "resources": {
        "limits": {
          "memory": "50M",
          "cpu": "100m"
          
        },
        "requests": {
          "memory": "50M",
          "cpu": "100m"
          
        }
      }
}
```
converts to:
```yml
resources:
  limits:
    memory: 50M
    cpu: 100m
  requests:
    memory: 50M
    cpu: 100m
```

Im not sure how this serialization would work but the needed outcome is the one above